src/bsp/src/stm32/src/TCPClient.cpp

```diff
@@ -1,7 +1,7 @@
 #include "TCPClient.h"
 
 TCPClient::TCPClient(int socket, sockaddr_in address, ILogger& logger) :
-    m_logger(logger), m_connected(socket > 0), m_socketFd(socket), m_address(address) {}
+    m_logger(logger), m_connected(socket >= 0), m_socketFd(socket), m_address(address) {}
```

Comment:
File descriptor 0 is stdin, why consider it connected for a socket?

Reply:
Because we aren't on UNIX and lwip doesn't use files. The return value of it's calls is nothing more than an index in an array. So the first started socket is at 0. Without this, the code just doesn't work.

Reply:
Good point